src/main/java/com/epam/grid/engine/provider/utils/slurm/healthcheck/ShowConfigCommandParser.java

```diff
@@ -0,0 +1,152 @@
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+import static com.epam.grid.engine.utils.TextConstants.SPACE;
+import static com.epam.grid.engine.utils.TextConstants.EMPTY_STRING;
+import static com.epam.grid.engine.utils.TextConstants.EQUAL_SIGN;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ShowConfigCommandParser {
+
+    private static final String UP_STATE = "UP";
+    private static final String DOWN_STATE = "DOWN";
+
+    private static final String SLURM_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
+    private static final String BOOT_TIME = "BOOT_TIME";
+    private static final String CONFIGURATION_STRING = "Configuration data as of";
+    private static final String SPACES_REGEX = "\\s+";
+    private static final Long NOT_PROVIDED = 99999L;
+
+    public static HealthCheckInfo parseShowConfigResult(final CommandResult commandResult) {
+        validateShowConfigResponse(commandResult);
+        final List<String> stdOut = commandResult.getStdOut();
+        final StatusInfo statusInfo = parseStatusInfo(stdOut);
+        return HealthCheckInfo.builder()
+                .statusInfo(statusInfo)
+                .startTime(getStartTime(stdOut))
+                .checkTime(getCheckTime(stdOut))
+                .build();
+    }
+
+    private static void validateShowConfigResponse(final CommandResult commandResult) {
+        if (CollectionUtils.isEmpty(commandResult.getStdOut())) {
+            throw new GridEngineException(HttpStatus.NOT_FOUND,
+                    String.format("Slurm error during health check. %nexitCode = %d %nstdOut: %s %nstdErr: %s",
+                            commandResult.getExitCode(), commandResult.getStdOut(), commandResult.getStdErr())
+            );
+        }
+        if (checkUnexpectedError(commandResult)) {
+            throw new GridEngineException(HttpStatus.INTERNAL_SERVER_ERROR,
+                    String.format("Unexcpected error occurred during health check. "
+                                    + "%nexitCode = %d %nstdOut: %s %nstdErr: %s",
+                            commandResult.getExitCode(), commandResult.getStdOut(), commandResult.getStdErr())
+            );
+        }
+    }
+
+    private static boolean checkUnexpectedError(final CommandResult commandResult) {
+        return commandResult.getExitCode() != 0 && !commandResult.getStdErr().isEmpty();
+    }
+
+    private static StatusInfo parseStatusInfo(final List<String> stdOut) {
+        final long statusCode = parseStatusCode(stdOut);
+        return StatusInfo.builder()
+                .code(statusCode)
+                .status(getStatus(statusCode))
+                .info(parseInfo(stdOut))
+                .build();
+    }
+
+    private static long parseStatusCode(final List<String> stdOut) {
+        final long statusCode;
+        final List<String> statusString = List.of(stdOut.get(stdOut.size() - 1)
```

Comment:
Let's handle the situation if `stdOut` is null or empty

Reply:
we check it in validateShowConfigResponse method
![image](https://user-images.githubusercontent.com/80297657/164526916-da88a2be-58a2-4940-907a-2529341be900.png)